### PR TITLE
Bug 1166214 - Make BaseModule.SUB_MODULES support subdirectory

### PR DESCRIPTION
**Pull request by luke-chang**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3956a7fc9f5dbc623b86c9d376e474f812e64b4c

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2c49c1be90e33e2ed8247b13005ebb86cdd004ca

File: apps/system/js/base_module.js [hunk 1]
```diff
@@ -415,20 +421,41 @@
...
 
   /**
+   * A helper function to split module expressions into "path" and "name".
+   * @example
+   * Service.parseModule('path/to/ModuleName');
+   * // {path: 'path/to/', name: 'ModuleName'}
+   * @param  {String} str String to be splitted
+   * @return {Object}     The result object with members: "path" and "name".
+   */
+  BaseModule.parseModule = function(str) {
```
Comment (alivedise) [inline apps/system/js/base_module.js]: nit: I think parse(Module)Path is better than parseModule

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6850b500038b82abe8038a359cc7c12b8ff14a75

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#1LtnkXuqQ26paSy69mcVvQ

The pull request failed to pass integration tests. It could not be landed, please try again.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ec9c6816d9272269bb82de4f51246140de7f95bb

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#ENqAiF6TTqy60lh1pArbAQ

The pull request failed to pass integration tests. It could not be landed, please try again.

**Pull request merged**